docs/gcp_workload_identity.md

```diff
@@ -0,0 +1,122 @@
+
+### Changes in the Credentials Secret with STS
+If we check the credentials secret, we have the following base64 encoded content in the `service_account.json` key of the `data` field.
+
+Without Workload Identity, the type of the credentials is `service_account`. These credentials include private RSA key, in `private_key` field, to be able to authenticate to gcp. This private key needs to kept secure and is not rotated.
+
+```json
+{
+   "type": "service_account",
+   "project_id": "test-project",
+   "private_key_id": "216fbde61a93dabd942ad738062f0ff0811b69f8",
+   "private_key": "<private_key>",
+   "client_email": "test-openshift-i-42ssv@test-project.iam.gserviceaccount.com",
+   "client_id": "100684158900346633229",
+   "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+   "token_uri": "https://oauth2.googleapis.com/token",
+   "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+   "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/test-service-account-42ssv@test-project.iam.gserviceaccount.com"
+}
+```
+
+With Workload Identity, the type of the credentials is `external_account`. `audience` is the target audience which is workload identity provider. `service_account_impersonation_url` key contains the resource url of the service account that can be impersonated with these credentials. `credentials_source.file` is the path to the oidc token, which is exchanged for google access token. The oidc token is rotated every one hour and thus credentials are short lived.
+
+```json
+{
+   "type": "external_account",
+   "audience": "//iam.googleapis.com/projects/123456789/locations/global/workloadIdentityPools/test-pool/providers/test-provider",
+   "subject_token_type": "urn:ietf:params:oauth:token-type:jwt",
+   "token_url": "https://sts.googleapis.com/v1/token",
+   "service_account_impersonation_url": "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/test-service-account-42ssv@test-project.iam.gserviceaccount.com:generateAccessToken",
+   "credential_source": {
+      "file": "/path/to/oidc/token",
+      "format": {
+         "type": "text"
+      }
+   }
+}
+```
+
+### Steps to install an OpenShift Cluster with Workload Identity
+
+1. Set $RELEASE_IMAGE to point to a sufficiently new OpenShift release
+2. Extract the GCP Credentials Request objects from the above release image. You must use 4.7 or newer version of `oc` CLI.
+   ```
+   mkdir credreqs ; oc adm release extract --cloud=gcp --credentials-requests $RELEASE_IMAGE --to=./credreqs ; cat ./credreqs/*.yaml > credreqs.yaml
```

Comment:
@akhil-rane For step 2, we don't need to make credreqs to one file, right?

Reply:
Nope, I will fix it. Nice catch!